vNext/UnityGLTFv2.cs

```diff
@@ -149,9 +159,34 @@ public interface IUnityGLTFExtension
 {
     IGLTFExtension GLTFExtension { get; };
 
-    Task<ExtensionReturnObject<GameObject>> CreateSceneAsync(GLTF.Schema.GLTFScene gltfScene, int indexToLoad);
-    Task<ExtensionReturnObject<GameObject>> CreateNodeAsync(Node gltfNode, int indexToLoad);
-    Task<ExtensionReturnObject<MeshPrimitive>> CreateMeshPrimitiveAsync(GLTFMesh mesh, ILoaderContext loaderContext, int indexToLoad);
+    /// <summary>
+    /// Creates a Unity object out of the glTF schema object
+    /// </summary>
+    /// <param name="importer">The importer that is used to load the object</param>
+    /// <param name="unityGLTFObject">Object that is being loaded</param>
+    /// <param name="sceneId">Index object which resolves to object in GLTFRoot</param>
+    /// <returns>The loaded glTF scene as a GameObject hierarchy</returns>
+    Task<ExtensionReturnObject<GameObject>> CreateSceneAsync(Importer importer, UnityGLTFObject unityGLTFObject, GLTF.Schema.SceneId sceneId);

```

Comment:
Why is the UnityGLTFObject passed in here? Is it so the extension can basically cache data in it? If so, then I guess this implies that the UnityGLTFObject is not opaque from the standpoint of the consumer of this API?

Reply:
Yes that UnityGLTFObject has the cache of data inside which may be needed for the extension

Reply:
That's right. The extension may need additional data from either the root or the asset cache.